qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -37,41 +43,80 @@ def _get_shared_links_for_study(study):
     """builds list of namedtuples for study listings"""
-    if studytype == "private":
-        studylist = user.user_studies
+    if studytype not in {"standard", "shared"}:
+        raise IncompetentQiitaDeveloperError("Must use standard or shared!")
+    # get list of studies for table
+    if studytype == "standard":
+        studylist = user.user_studies.union(Study.get_by_status('public'))
     elif studytype == "shared":
         studylist = user.shared_studies
-    elif studytype == "public":
-        studylist = Study.get_by_status('public')
-    else:
-        raise IncompetentQiitaDeveloperError("Must use private, shared, "
-                                             "or public!")
-
-    StudyTuple = namedtuple('StudyInfo', 'id title meta_complete '
-                            'num_samples_collected shared num_raw_data pi '
-                            'pmids owner status abstract')
+    if results is not None:
+        studylist = studylist.intersection(results)
+    if not studylist:
+        # No studies left so no need to continue
+        return []
+
+    # get info for the studies
+    cols = ['study_id', 'status', 'email', 'principal_investigator_id',
+            'pmid', 'study_title', 'metadata_complete',
+            'number_samples_collected', 'study_abstract']
+    study_info = Study.get_info(studylist, cols)
 
     infolist = []
-    for s_id in studylist:
-        study = Study(s_id)
+    for row, info in enumerate(study_info):
+        study = Study(info['study_id'])
         status = study.status
         # Just passing the email address as the name here, since
         # name is not a required field in qiita.qiita_user
-        owner = study_person_linkifier((study.owner, study.owner))
-        info = study.info
+        owner = study_person_linkifier((info['email'], info['email']))
         PI = StudyPerson(info['principal_investigator_id'])
         PI = study_person_linkifier((PI.email, PI.name))
-        pmids = ", ".join([pubmed_linkifier([pmid])
-                           for pmid in study.pmids])
+        if info['pmid'] is not None:
+            pmids = ", ".join([pubmed_linkifier([p])
+                               for p in info['pmid']])
+        else:
+            pmids = ""
+        if info["number_samples_collected"] is None:
+            info["number_samples_collected"] = "0"
         shared = _get_shared_links_for_study(study)
-        infolist.append(StudyTuple(study.id, study.title,
-                                   info["metadata_complete"],
-                                   info["number_samples_collected"],
-                                   shared, len(study.raw_data()),
-                                   PI, pmids, owner, status,
-                                   info["study_abstract"]))
+        meta_complete_glyph = "ok" if info["metadata_complete"] else "remove"
+        # build the HTML elements needed for table cell
+        title = ("<a href='#' data-toggle='modal' "
```

Comment:
This is all in here because it's returned as part of the json that is used to fill the table. Therefore you need the full column data to put in each cell.